ui/admin/app/components/form/role/manage-custom-scopes/index.js

```diff
@@ -30,13 +27,14 @@ export default class FormRoleManageCustomScopesIndexComponent extends Component
   selectionChange({ selectableRowsStates }) {
     selectableRowsStates.forEach((row) => {
       const { isSelected, selectionKey: key } = row;
-      const includesId = this.selectedItems.includes(key);
+      // If index is equal to -1 then key does not exist in the array.
+      const index = this.args.selectedItems.indexOf(key);
       if (isSelected) {
-        if (!includesId) this.selectedItems = [...this.selectedItems, key];
-      } else if (includesId) {
-          this.selectedItems = this.selectedItems.filter(
-            (item) => item !== key,
-          );
+        if (index === -1) {
```

Comment:
```
if (isSelected && index === -1) {
  this.args.selectedItems.push(key);
} else if (!isSelected && index !== -1) {
  this.args.selectedItems.splice(index, 1);
}
```

I think either way works but I think this might be a bit easier to read? It duplicates the check for `isSelected` but sometimes that makes it easier to understand, I also at first glance thought the same thing @cameronperera did

Reply:
great point! updated!